utils/image_io/src/image_io.cpp

```diff
@@ -165,8 +165,8 @@ torch::Tensor load_image(const std::string& file_path,
 
     std::vector<unsigned char> image_resized_buffer(buffer_size);
 
-    stbir_resize_uint8(image_raw.get(), width, height, 0,
-        image_resized_buffer.data(), new_width, new_height, 0, depth);
+    stbir_resize_uint8_linear(image_raw.get(), width, height, 0,
+        image_resized_buffer.data(), new_width, new_height, 0, STBIR_RGBA);
```

Comment:
Please use `static_cast<stbir_pixel_layout>(depth)` instead of hard-coding `STBIR_RGBA` here - otherwise correct resizing fails for images with != 4 channels. This is recommended in the mentioned guide and the `stb` author also casts the channel count parameter to `stbir_pixel_layout` in tests (see
https://github.com/nothings/stb/blob/beebb24b945efdea3b9bba23affb8eb3ba8982e7/tests/resample_test.cpp#L146).

Reply:
That makes sense, just fixed it!